code/Model/EditableFormField.php

```diff
@@ -1003,6 +991,48 @@ public function formatDisplayRules()
         return (count($result['selectors'])) ? $result : null;
     }
 
+    /**
+     * Check if this EditableFormField is displayed based on its DisplayRules and the provided data.
+     * @param array $data
+     * @return bool
+     */
+    public function isDisplayed(array $data)
+    {
+        $displayRules = $this->DisplayRules();
+
+        if ($displayRules->count() === 0) {
+            // If no display rule have been defined, isDisplayed equals the ShowOnLoad property
+            return $this->ShowOnLoad;
+        }
+
+        $conjunction = $this->DisplayRulesConjunctionNice();
+
+        // && start with true and find and condition that doesn't satisfy
+        // || start with false and find and condition that satisfies
+        $conditionsSatisfied = ($conjunction === '&&');
+
+        foreach ($displayRules as $rule) {
+            $controllingField = $rule->ConditionField();
+
+            // recursively check - if any of the dependant fields are hidden, assume the rule can not be satisfied
+            $ruleSatisfied = $controllingField->isDisplayed($data) && $rule->validateAgainstFormData($data);
+
+            if ($conjunction === '||' && $ruleSatisfied) {
+                $conditionsSatisfied = true;
+                break;
+            }
+            if ($conjunction === '&&' && !$ruleSatisfied) {
+                $conditionsSatisfied = false;
+                break;
+            }
+        }
+
+        // initially displayed - condition fails || initially hidden, condition passes
+        $startDisplayed = $this->ShowOnLoad;
+        return ($startDisplayed xor $conditionsSatisfied);
```

Comment:
Most people won't know what `xor` does (I have to look it up myself), so would probably just make it long form.  Also don't need the $startDisplayed variable.

`return ($this->ShowOnLoad || $conditionsSatisfied) && $this->ShowOnLoad != $conditionsSatisfied;`

Reply:
Dammit Steve, I almost never come across a legitimate usecase of the exclusive disjunction operator and now that I've found one you would deny the satisfaction of using it!!! :wink: 

On a serious note, I don't think `xor` is such an obscure operator. And I it communicates the important point that if both `$startDisplayed` and `$conditionsSatisfied` are true, that `isDisplayed` should return `false`.

If you throw this statement at me and ask me what will be return if both condition are true, I'll have to think about it for a second before giving you an answer.
```
return ($this->ShowOnLoad || $conditionsSatisfied) && $this->ShowOnLoad != $conditionsSatisfied;
```

Reply:
hahaha :-)  keep the xor then ❌ ⭕️ 🐰